packages/kbn-test/src/functional_tests/lib/run_kibana_server.js

```diff
@@ -59,12 +59,35 @@ function collectCliArgs(config, { installDir, extraKbnOpts }) {
     serverArgs,
     args => (installDir ? args.filter(a => a !== '--oss') : args),
     args => {
-      return installDir ? [...args, ...buildArgs] : [KIBANA_EXEC_PATH, ...args, ...sourceArgs];
+      return installDir ? [...buildArgs, ...args] : [KIBANA_EXEC_PATH, ...sourceArgs, ...args];
```

Comment:
`args` refers to all the options that the user passes in via command line and/or the particular test config.js they're using for starting the Kibana test server.

Previously we were appending `sourceArgs` (or `buildArgs` if they passed in `--kibana-install-dir`) to the end of their passed-in `args`. But this doesn't work when they're using `--server.basePath` because `sourceArgs` was always overriding it with `--no-base-path` option. In fact for base path settings, the ordering doesn't matter because `no-base-path` and `server.basePath` are different keys; just the _existence_ of `no-base-path` disables the base path proxy server.

Now, I've flipped the ordering so that the user's `args` always override the default build/source args. And, I've made sure that any base path setting that comes later, whether or not the keys match, will override previous base path settings.